### DISCUSSION TITLE
Provide some predefined common layout templates

### DISCUSSION THREAD

**Discussion by qingant** [zauberzeug/nicegui#1461] [Ideas / Feature Requests]
I come from Streamlit because it would never be able to scale to medium to large sized web app. But the fact that you would have a nice page without any effort when you use Streamlit is great. And that's a big reason of Streamlit's success.

I think we can implement some pre defined template that defines the layout and style, and the interface could be
```py
ui.run(template='xxx')
```

**Comment by falkoschindler** [comment]
Interesting idea, @qingant!
Can you give a specific example for such a template? What would some dummy code and the resulting layout look like?

**Reply by v479038280** [reply]
Here is an [example ](https://panel.holoviz.org/getting_started/core_concepts.html),
>Templates
Once you have started to build an application, you will probably want to make it look nicer, which is where templates come in. Whenever you mark an object as .servable you are inserting it into a template. By default Panel uses a completely blank template, but it is very simple to select another template by setting pn.config.template. Here you will have a few options based on different frameworks, including 'bootstrap', 'material' and 'fast'.

**Reply by usekevinwu** [reply]
hi ,I encountered the same need，i understand it.  Quasar Layout Gallery https://quasar.dev/layout/gallery
